collector/processor/batchprocessor/batch_processor.go

```diff
@@ -26,8 +27,11 @@ import (
 	"go.opentelemetry.io/collector/processor"
 )
 
-// errTooManyBatchers is returned when the MetadataCardinalityLimit has been reached.
-var errTooManyBatchers = consumererror.NewPermanent(errors.New("too many batcher metadata-value combinations"))
+var (
+	// errTooManyBatchers is returned when the MetadataCardinalityLimit has been reached.
+	errTooManyBatchers = consumererror.NewPermanent(errors.New("too many batcher metadata-value combinations"))
+	errTimedOut = errors.New("processing items timed out")
```

Comment:
Note: It looks to me like a non-error condition below, where you used this variable. Maybe we can remove it, let's discuss.